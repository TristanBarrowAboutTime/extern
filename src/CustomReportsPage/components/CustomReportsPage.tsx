import * as React from 'react';
import FolderSelectionList from './folder-selection-list/FolderSelectionList';
import Modal from './molecular-components/Modal';
import styled from 'styled-components';
import { useCustomReportsPage } from '../hooks/component-hooks/useCustomReportsPage';
import FolderView from './folders-view/FolderView';

const Container = styled.div`
    display: flex;
    flex-direction: row;
    width: 100%;
`;

const CustomReportsPage = () => {
    const binding = useCustomReportsPage();
    console.log(binding.selectionBinding)
    return (
        <Container>
            {binding.isShowingModal && <Modal 
                title={'Confirm Delete'} 
                content={'Are you sure you want to delete this folder?'}
                buttons={binding.modalButtons}
                closeModal={binding.closeModal}
            />}
            <FolderSelectionList {...binding.selectionBinding}/>
            <FolderView
                folderName={binding.folderName}
            />
        </Container>
    );
}

export default CustomReportsPage;