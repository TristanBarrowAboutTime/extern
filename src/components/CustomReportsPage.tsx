import React from 'react';
import FolderSelectionList from './folder-selection-list/FolderSelectionList';
import Modal from './molecular-components/Modal';
import styled from 'styled-components';
import { useCustomReportsPage } from '../hooks/component-hooks/useCustomReportsPage';

const Container = styled.div`
    display: flex;
    flex-direction: row;
    width: 100%;
    background-color: #E5E5E5;
`;

export const CustomReportsPage = () => {
    const binding = useCustomReportsPage();
    
    return (
        <Container>
            {binding.isShowingModal && <Modal 
                title={'Confirm Delete'} 
                content={'Are you sure you want to delete this folder?'}
                buttons={binding.modalButtons}
                closeModal={binding.closeModal}
            />}
            <FolderSelectionList {...binding.selectionBinding}/>
            <div>{"folders view"}</div>
        </Container>
    );
}

export default CustomReportsPage;