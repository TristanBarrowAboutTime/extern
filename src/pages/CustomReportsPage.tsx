import * as React from 'react';
import FolderSelectionList from '../components/cellular-components/folder-selection-list/FolderSelectionList';
import Modal from '../components/molecular-components/Modal';
import styled from 'styled-components';
import { useCustomReportsPage } from '../hooks/component-hooks/useCustomReportsPage';
import FolderView from '../components/cellular-components/folders-view/FolderView';

const Container = styled.div`
    display: flex;
    flex-direction: row;
    width: 100%;
`;

const CustomReportsPage = () => {
    const binding = useCustomReportsPage();
    
    return (
        <Container>
            {binding.isShowingModal && <Modal 
                title={'Confirm Delete'}
                content={'Are you sure you want to delete this folder?'}
                buttons={binding.modalButtons}
                closeModal={binding.closeModal}
            />}
            <FolderSelectionList {...binding.selectionBinding}/>
            <FolderView
                folderName={binding.folderName}
                folders={binding.selectionBinding.folders}
            />
        </Container>
    );
}

export default CustomReportsPage;