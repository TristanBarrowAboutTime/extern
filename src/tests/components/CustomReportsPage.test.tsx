import React from 'react';
import renderer from 'react-test-renderer';
import CustomReportsPage from '../../components/CustomReportsPage';

// just a place holder
describe('Custom Reports Page', () => {
    it('renders Correctly', () => {
        const component = renderer.create(
            <CustomReportsPage />
        ).toJSON();
        expect(component).toMatchSnapshot();

    })
})