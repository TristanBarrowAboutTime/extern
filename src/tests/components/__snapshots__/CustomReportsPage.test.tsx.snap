// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Reports Page renders Correctly 1`] = `
<div
  className="sc-hiKfjK csceFZ"
>
  <div
    className="sc-dIvqjp GXvTt"
  >
    <div
      className="sc-hHEjAm iyJgp"
    >
      <div
        className="sc-fKgIGh bFPAGI"
      >
        Folders
      </div>
      <div
        className="sc-iwaifL jgzifN"
      >
        <button
          className="sc-bdnylx iUsuKv"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
        >
          New Folder
        </button>
        <button
          className="sc-bdnylx dYkfvg"
          disabled={false}
          onClick={[Function]}
          style={Object {}}
        >
          Reorder
        </button>
      </div>
    </div>
    <input
      className="sc-cxNIbT llUZnM"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <div
      className="sc-dlMBXb bZPHkC"
    >
      <div
        className="sc-kfYqjs"
      >
        <div
          className="sc-ksluoS iijERe"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            All Reports
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Favorites
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Recent
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Created by Me blah blah blah blah blah blah blah blah
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Shared
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Scheduled
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Job Reports
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Productivity Reports
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Created by Me2
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Shared2
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Scheduled2
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Job Reports2
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Productivity Reports2
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Shared3
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Scheduled3
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Job Reports3
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Productivity Reports3
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Shared4
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Scheduled4
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Job Reports4
          </div>
        </div>
        <div
          className="sc-ksluoS gNxkFq"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-hBMVcZ hiJaYW"
          >
            Productivity Reports4
          </div>
        </div>
        <hr
          className="sc-bCwgka idxTGg"
        />
        <div
          style={
            Object {
              "color": "#777777",
              "marginLeft": 32,
            }
          }
        >
          Recently Deleted
        </div>
        <div />
      </div>
    </div>
  </div>
  <div>
    folders view
  </div>
</div>
`;
